tests/functional/tests.py

```diff
@@ -16,37 +16,56 @@
 # limitations under the License.
 
 import os
-import io
 import uuid
-import urllib3
-import certifi
 import shutil
+from random import random
 
-from random import choice
 from string import ascii_lowercase
-
 from datetime import datetime, timedelta
 
+import urllib3
+import certifi
+
 from minio import Minio, PostPolicy, CopyConditions
 from minio.policy import Policy
 from minio.error import (ResponseError, PreconditionFailed,
                          BucketAlreadyOwnedByYou, BucketAlreadyExists)
 
-class RandomData:
+class LimitedRandomReader(object):
     """
-    RandomData quickly generates pseudo random data by chunks of 1024
+    LimitedRandomReader returns a Reader that upon read
+    returns random data, but stops with EOF after *limit*
+    bytes.
+
+    LimitedRandomReader is compatible with BufferedIOBase.
+
+    returns a class:`LimitedRandomReader` that upon read
+       provides random data and stops with EOF after *limit*
+       bytes
+
+    :param limit: Trigger EOF after limit bytes.
     """
-    def __init__(self, choices_list, randomness_level):
-        self.block = 1024
-        self.seed = []
-        for _ in range(randomness_level):
-            self.seed.append(''.join([choice(choices_list) for _ in range(self.block)]))
-
-    def gen(self, nr_bytes):
-        """ generates random data for nr_bytes length """
-        if nr_bytes % self.block is not 0:
-            raise ValueError('Passed number of bytes should be multiple of ' + self.block)
-        return ''.join([choice(self.seed) for _ in range(int(nr_bytes/self.block))])
+    def __init__(self, limit):
+        self._limit = limit
+        self._offset_location = 0
+
+    def read(self, amt=64*1024):
+        """
+        Similar to :meth:`io.read`, with amt option.
+
+        :param amt:
+            How much of the content to read.
+        """
+        # If offset is bigger than size. Treat it as EOF return here.
+        if self._offset_location + amt > self._limit:
```

Comment:
we should return empty bytes (`b''`) only when `self._offset_location == self._limit`. E.g, when
`self._limit = 10`, `self._offset = 9` and `amt = 5`, we should return the last byte and not `b''`.